### PR TITLE
Xiopak upgrade and error test improvements

### PR DESCRIPTION
**Pull request by galanCA**
## Description

Upgrades xiopak.cpp to be use in all the systems. The changes performed were:
- Delete direct library and change the slash for types.h
- Replace unlink for std::remove
- Replace splitpath and other windows functions for a c++ standard using filesystem.
- Add a better test suite for error and error code checking

## Test

A new ctest was added to test the logic of not-found-file.

## Author Progress Checklist:

- [ ] Open draft pull request
    - [ ] Make title clearly understandable in a standalone change log context
    - [ ] Assign yourself the issue
    - [ ] Add at least one label (enhancement, bug, or maintenance)
    - [ ] Link the issue(s) addressed by this PR (under "Development" in the sidebar menu) 
- [ ] Make code changes (if you haven't already)
- [ ] Self-review of code
    - [ ] My code follows the style guidelines of this project
    - [ ] I have added comments to my code, particularly in hard-to-understand areas
    - [ ] I have only committed the necessary changes for this fix or feature
    - [ ] I have made corresponding changes to the documentation
    - [ ] My changes generate no new warnings
    - [ ] I have ensured that my fix is effective or that my feature works as intended by:
        - [ ] exercising the code changes in the test framework, and/or
        - [ ] manually verifying the changes (as explained in the the pull request description above)
    - [ ] My changes pass all local tests
    - [ ] My changes successfully passes CI checks
    - [ ] Add any unit test for proof and documentation.
    - [ ] Merge in main branch and address resulting conflicts and/or test failures.
- [ ] Move pull request out of draft mode and assign reviewers
- [ ] Iterate with reviewers until all changes are approved
    - [ ] Make changes in response to reviewer comments
    - [ ] Merge in main branch and address resulting conflicts and/or test failures.
    - [ ] Re-request review in GitHub

## Reviewer Checklist:

 - [ ] Read the pull request description
 - [ ] Perform a code review on GitHub
 - [ ] Confirm all CI checks pass and there are no build warnings
 - [ ] Pull, build, and run automated tests locally
 - [ ] Perform manual tests of the fix or feature locally
 - [ ] Add any review comments, if applicable
 - [ ] Submit review in GitHub as either
     - [ ] Request changes, or
     - [ ] Approve
 - [ ] Iterate with author until all changes are approved

### CODE REVIEW

**Comment by nealkruis** [comment]
This is waiting on a review from @tanaya-bigladder, but it does not describe or add any tests.

File: src/xiopak.cpp
Comment (tanaya-bigladder) [inline src/xiopak.cpp:603]: This all looks reasonable, but it's very time-consuming (i.e. doubles the work done) if the reviewer has to verify that all the new code performs the same logic as the old code. Isn't there a place where we can put tests for these functions now?

**Review by tanaya-bigladder — CHANGES_REQUESTED** [review]

File: src/xiopak.cpp
Comment (galanCA) [reply src/xiopak.cpp:603]: I added a new test for non-found file logic.

File: test/error_handling/ref/no_file.err
Comment (galanCA) [inline test/error_handling/ref/no_file.err:3]: @nealkruis Do I create this error file in cmake and add it to gitignore?

File: src/xiopak.cpp
Comment (tanaya-bigladder) [reply src/xiopak.cpp:603]: Did you commit `no_file.cse`?

Comment (galanCA) [reply src/xiopak.cpp:603]: This was solved in slack. No `no_file.cse` is need it.

File: cmake/string-finder.cmake
Comment (nealkruis) [inline cmake/string-finder.cmake:8]: These are not very informative messages.

**Review by nealkruis — CHANGES_REQUESTED** [review]

**Comment by galanCA** [comment]
The changes from the last commit were done from a pair programming session with @nealkruis.

**Review by nealkruis — APPROVED** [review]

**Review by tanaya-bigladder — APPROVED** [review]

**Pull request merged**
